### PR TITLE
Candidatetestpageapiconnect

### PR DESCRIPTION
**Pull request by hnalabanda**
There were API changes made for this change. Please get the API changes from the PR for Api
Currently on testpage.tsx the token has been hard coded to 66666
-So the first time you will be allowed to submit the test 
-When you try to submit the next time you will be shown an error

### CODE REVIEW

File: components/TextEditor/TextEditorContainer.tsx [hunk 1]
```diff
@@ -36,12 +66,12 @@ const TextEditorContainer: FunctionComponent<TextEditorContainerProps> = ({heigh
                 options={TextEditorSettings}
             />
         </div>
-            <Link href={"/submitted"}>
-                <a className={scss.buttonBlack} onClick={handleShowValue}>Submit Code</a>
-            </Link>
+            
+                <a className={scss.buttonBlack} onClick={submitForm}>Submit Code</a>
```
Comment (MichaelWalker) [inline components/TextEditor/TextEditorContainer.tsx:70]: Could you change this to a button?

File: components/TextEditor/TextEditorContainer.tsx [hunk 2]
```diff
@@ -1,30 +1,60 @@
...
+        setToken(token);
     }
 
-    function handleShowValue(): void {
-        alert(`You have submitted this code: ${getEditorContentIfMountedRef.current()}`);
+   
+    function submitForm() {      
+        addTestSubmisson(tokenId,{testId,testAnswer})
+            .then((response)=>{  
+                if (response.status>= 200 && response.status <= 299) {
```
Comment (MichaelWalker) [inline components/TextEditor/TextEditorContainer.tsx:43]: response already has a nice method for this! :)

You can just do `response.ok`

File: components/TextEditor/TextEditorContainer.tsx [hunk 3]
```diff
@@ -1,30 +1,60 @@
...
     defaultText: string;
+    token:string;
+    testNumber:number
 }
-const TextEditorContainer: FunctionComponent<TextEditorContainerProps> = ({height, width, defaultText}) => {
+const TextEditorContainer: FunctionComponent<TextEditorContainerProps> = ({height, width, defaultText,token,testNumber}) => {
     const [isEditorReady, setIsEditorReady] = useState(false);
+    const [status, setStatus] = useState<FormStatus>("READY");
+    const [tokenId, setToken] = useState("");
+    const [testId, setTest] = useState(0);
```
Comment (MichaelWalker) [inline components/TextEditor/TextEditorContainer.tsx:25]: I don't think we need state for these?
Could we just use the value from the props?

File: components/TextEditor/TextEditorContainer.tsx [hunk 4]
```diff
@@ -1,30 +1,60 @@
...
     height: string;
     width: string;
     defaultText: string;
+    token:string;
+    testNumber:number
 }
-const TextEditorContainer: FunctionComponent<TextEditorContainerProps> = ({height, width, defaultText}) => {
+const TextEditorContainer: FunctionComponent<TextEditorContainerProps> = ({height, width, defaultText,token,testNumber}) => {
     const [isEditorReady, setIsEditorReady] = useState(false);
+    const [status, setStatus] = useState<FormStatus>("READY");
```
Comment (MichaelWalker) [inline components/TextEditor/TextEditorContainer.tsx:23]: Do we ever use this?

File: components/TextEditor/TextEditorContainer.tsx [hunk 5]
```diff
@@ -1,30 +1,60 @@
...
+    const [testAnswer, setTestAnswer] = useState("");
+    const [error, setError] = useState("");
     const getEditorContentIfMountedRef: MutableRefObject<EditorContentGetter> = useRef(() => "");
-
+    const router=useRouter();
+    
     function handleIsEditorMounted(_getEditorContents: EditorContentGetter): void {
         setIsEditorReady(true);
         getEditorContentIfMountedRef.current = _getEditorContents;
+        setTestAnswer(getEditorContentIfMountedRef.current());
```
Comment (MichaelWalker) [inline components/TextEditor/TextEditorContainer.tsx:34]: won't this just set the `testAnswer` to whatever it is when you open the page?
How does it update to get the new value when you are ready to submit?

File: pages/api/candidateApiClient.module.ts [hunk 6]
```diff
@@ -36,3 +41,18 @@ export async function getCandidateTests() {
...
+  
+        const response = await fetch(`${apiURL}/sessions/${tokenId}`, {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify(newTestSubmission),
+        });
+    return await response;
+
```
Comment (MichaelWalker) [inline pages/api/candidateApiClient.module.ts:57]: nice! :)

File: components/TextEditor/TextEditorContainer.tsx (hunk 1 above)
Comment (hnalabanda) [reply components/TextEditor/TextEditorContainer.tsx:70]: Changed to a button

File: components/TextEditor/TextEditorContainer.tsx (hunk 2 above)
Comment (hnalabanda) [reply components/TextEditor/TextEditorContainer.tsx:43]: changed to use response.ok

File: components/TextEditor/TextEditorContainer.tsx (hunk 3 above)
Comment (hnalabanda) [reply components/TextEditor/TextEditorContainer.tsx:25]: removes usestates and am now using props

File: components/TextEditor/TextEditorContainer.tsx (hunk 4 above)
Comment (hnalabanda) [reply components/TextEditor/TextEditorContainer.tsx:23]: No,I don't use it . Have removed them

File: components/TextEditor/TextEditorContainer.tsx (hunk 5 above)
Comment (hnalabanda) [reply components/TextEditor/TextEditorContainer.tsx:34]: getting the getEditorContentIfMountedRef.current() in the submitForm method now

**Pull request merged**